### PR TITLE
ARUHA-527: tests for timelines CRUD

### PR DESCRIPTION
**Pull request by adyach**
Added tests for controller and service

### CODE REVIEW

File: src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    private final TopicRepositoryHolder topicRepositoryHolder = Mockito.mock(TopicRepositoryHolder.class);
+    private final TransactionTemplate transactionTemplate =
+            new TransactionTemplate(Mockito.mock(PlatformTransactionManager.class));
+
+    private final TimelineService timelineService = new TimelineService(securitySettings, eventTypeCache,
+            storageDbRepository, timelineSync, nakadiSettings, timelineDbRepository, topicRepositoryHolder,
+            transactionTemplate, new UUIDGenerator());
+
+    @Test
+    public void testCreateTimelineFromFake() throws Exception {
```
Comment (rcillo) [inline src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java]: This test has too many mocks. I think it would be better just to test it in the "acceptance tests".

File: src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        timelineService.createTimeline("event_type", "storage_id", new FullAccessClient("clientId"));
+
+        Mockito.verify(timelineDbRepository).createTimeline(Matchers.any());
+        Mockito.verify(timelineSync).startTimelineUpdate(Matchers.eq(eventType.getName()), Matchers.anyLong());
+        Mockito.verify(timelineDbRepository).updateTimelime(Matchers.any());
+        Mockito.verify(timelineSync).finishTimelineUpdate(eventType.getName());
+    }
+
+    @Test
+    public void testCreateTimelineFromReal() throws Exception {
```
Comment (rcillo) [inline src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java]: The same for this one. Maybe acceptance tests suites better for this scenario.

File: src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        Mockito.verify(timelineDbRepository).createTimeline(Matchers.any());
+        Mockito.verify(timelineSync)
+                .startTimelineUpdate(Matchers.eq(activeTimeline.getEventType()), Matchers.anyLong());
+        Mockito.verify(timelineDbRepository).updateTimelime(Matchers.eq(activeTimeline));
+        Mockito.verify(timelineDbRepository).updateTimelime(Matchers.eq(activeTimeline));
+        Mockito.verify(timelineSync).finishTimelineUpdate(activeTimeline.getEventType());
+    }
+
+    @Test
+    public void testDeleteTimeline() throws Exception {
```
Comment (rcillo) [inline src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java]: The same for this.

File: src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+        timelineService.delete(eventType.getName(), timelineId, new FullAccessClient("clientId"));
+
+        Mockito.verify(timelineSync).startTimelineUpdate(Matchers.eq(eventType.getName()), Matchers.anyLong());
+        Mockito.verify(timelineDbRepository).deleteTimeline(UUID.fromString(timelineId));
+        Mockito.verify(timelineSync).finishTimelineUpdate(eventType.getName());
+    }
+
+    @Test
+    public void testGetTimelines() throws Exception {
```
Comment (rcillo) [inline src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java]: The same.

File: src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        Mockito.when(eventTypeCache.getEventType(Matchers.any())).thenReturn(eventType);
+        Mockito.when(timelineDbRepository.listTimelines(eventType.getName())).thenReturn(timelines);
+
+        final List<Timeline> actualTimelines =
+                timelineService.getTimelines(eventType.getName(), new FullAccessClient("clientId"));
+        Assert.assertEquals(timelines, actualTimelines);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void testGetTimelinesNotFound() throws Exception {
```
Comment (rcillo) [inline src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java]: This one is fine. I think that main scenarios like create from fake and create real could be done only in acceptance tests. Then, corner case scenarios could be exercised here through unit tests with small number of mocks.

**Comment by rcillo** [comment]
@adyach it would be interesting to clean up some tests relying mostly on mocks and move them to acceptance tests. Maybe we could add timelines switching steps to the UserJourney? Or even have a new "Timeline User Journey" scenario that creates from fake, delete, creates from real, all in a single run, for performance reasons.

**Comment by adyach** [comment]
@rcillo I have moved the heavily mocked tests to acceptance tests. Please, check it out.

**Comment by rcillo** [comment]
@adyach looks good 👍 

I think it just needs minor fixes in tests. Probably removing timelines on `setUp` method.

**Comment by rcillo** [comment]
:shipit:
